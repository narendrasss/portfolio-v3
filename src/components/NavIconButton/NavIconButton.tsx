import React from 'react';
import Button, { ButtonProps } from '../base/Button';
import styled from 'styled-components';

const Container = styled(Button)`
  width: 16px;
  height: 16px;
  background: black;
` as React.FC<ButtonProps>;

const NavIconButton: React.FC<ButtonProps> = props => (
  <Container data-testid="button" {...props} />
);

export default NavIconButton;
