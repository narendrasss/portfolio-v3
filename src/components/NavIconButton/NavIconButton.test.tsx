import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import NavIconButton from './NavIconButton';

describe('NavIconButton', () => {
  const mockHandler = jest.fn();

  it('renders without crashing', () => {
    const { asFragment } = render(<NavIconButton onClick={mockHandler} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls click handler when clicked', () => {
    const { getByTestId } = render(<NavIconButton onClick={mockHandler} />);
    fireEvent.click(getByTestId('button'));
    expect(mockHandler).toHaveBeenCalled();
  });
});
