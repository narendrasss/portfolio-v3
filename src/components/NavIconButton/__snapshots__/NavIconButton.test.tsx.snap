// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavIconButton renders without crashing 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa sc-dnqmqq hLixnB sc-htoDjs qIBAg"
    data-testid="button"
    type="button"
  />
</DocumentFragment>
`;
